libs/calculate-deps.lua

```diff
@@ -176,10 +178,10 @@ function processDeps(dependencies)
   end
 end
 
-return function (gitDb, depsMap, newDeps)
+return function (gitDb, depsMap, newDeps, core)
```

Comment:
Is there a reason why you pass the whole core module here instead of just the `config` field?

Reply:
The only reason is future-proofing the needs of calculate-deps, as of now it can be replaced with config. Should I prefer doing that or just keep passing core assuming in some future PR it would be needed for something else?

It also seemed to me like the rest of stuff in core.lua passes the entire core even though only some single property being used.

Reply:
I'm not really sure. It's just the first thing I noticed when I started reviewing this and I wanted to clarify before moving forward.

Reply:
It seems fine to me so I will keep it for now.

Feel free to ask as many questions, I will do my best answering them.

Reply:
Looking at this code again, I think we should change this.  Because I also noticed I am not passing `core` in the 3 other calls to calculate-deps + having to pass `core.db`.  I think it would look better if first argument was simply the core, and we use that to get the db and the configs.  I suppose such change wouldn't be considered breaking since calculate-deps is not exposed to end-user.